pkg/option/config.go

```diff
@@ -3638,26 +3638,36 @@ func (c *DaemonConfig) checkMapSizeLimits() error {
 }
 
 func (c *DaemonConfig) checkIPv4NativeRoutingCIDR() error {
-	if c.GetIPv4NativeRoutingCIDR() == nil && c.EnableIPv4Masquerade && !c.TunnelingEnabled() &&
-		c.IPAMMode() != ipamOption.IPAMENI && c.EnableIPv4 && c.IPAMMode() != ipamOption.IPAMAlibabaCloud {
+	if c.GetIPv4NativeRoutingCIDR() == nil &&
+		c.EnableIPv4 && c.EnableIPv4Masquerade &&
```

Comment:
Nit: We could remove `c.EnableIPv4` here, given that `c.EnableIPv4Masquerade` already depends on it (same below for IPv6), see earlier in the file:

```go
     c.EnableIPv4Masquerade = vp.GetBool(EnableIPv4Masquerade) && c.EnableIPv4
     c.EnableIPv6Masquerade = vp.GetBool(EnableIPv6Masquerade) && c.EnableIPv6
```

Reply:
Good point, I didn't realize that. However, since we do print `EnableIPv{4,6}` in the error message, I think I'll leave it as is. That way, the if condition and the error message stay in sync.

Reply:
Sounds good to me :+1: